src/sqlancer/mongodb/MongoDBProvider.java

```diff
@@ -72,8 +72,10 @@ protected void executeEpilogue(Query<?> q, boolean success, ExecutionTimer timer
             if (success && getOptions().printSucceedingStatements()) {
                 System.out.println(q.getLogString());
             }
-            if (logExecutionTime) {
-                getLogger().writeCurrent("// " + timer.end().asString());
+            if (getOptions().logEachSelect()) {
```

Comment:
I think the logic would be slightly easier to understand if you have the logic in one if statement (here and above): `if (getOptions().logEachSelect() && logExecutionTime)`

Reply:
Sure! Fixed in a newer commit.